### PR TITLE
Add [LGTM] service

### PR DESCRIPTION
**Pull request by samlanning**
This PR adds the lgtm.com alerts count for a particular project, using the public-facing API.

Full disclosure, I work at Semmle on lgtm.com.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1067,6 +1067,43 @@ cache(function(data, match, sendBadge, request) {
   }
 }));
 
+// LGTM alerts integration
+camp.route(/^\/lgtm\/alerts\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var projectId = match[1]; // eg, `g/apache/cloudstack`
+  var format = match[2];
+  var url = 'https://lgtm.com/api/v0.1/project/' + projectId + '/details';
+  var badgeData = getBadgeData('lgtm', data);
```
Comment (chris48s) [inline server.js]: Could you use the ES6 style `let` and `const` (I know we have a lot of code using `var`, but we're trying to update as we add new code).

File: server.js [hunk 2]
```diff
@@ -1067,6 +1067,43 @@ cache(function(data, match, sendBadge, request) {
...
+  request(url, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var data = JSON.parse(buffer);
+      if (data.code === 404) {
```
Comment (chris48s) [inline server.js]: Assuming when `data.code === 404` the HTTP status code is also 404, this block here and the section above where you're checking for `if (err != null)` can all be simplified to:

```js
if (checkErrorResponse(badgeData, err, res, 'project not found')) {
  sendBadge(format, badgeData);
  return;
}
```

it is a very common check, so we have a [helper](https://github.com/badges/shields/blob/3cb71037fdbed20f74a18880ed9edad4273a08c5/lib/error-helper.js) for it

File: services/lgtm/lgtm.tester.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const ServiceTester = require('../service-tester');
+
+const t = new ServiceTester({ id: 'lgtm', title: 'LGTM' })
+module.exports = t;
+
+t.create('total alerts for a project')
+  .get('/alerts/g/apache/cloudstack.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'lgtm',
+    value: Joi.string().regex(/^[0-9kM.]+ alerts?$/)
```
Comment (chris48s) [inline services/lgtm/lgtm.tester.js]: I think this regex should allow 'alert' or 'alerts' (i.e: it should pass however many alerts `apache/cloudstack` has when we run the test).

File: services/lgtm/lgtm.tester.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  }));
+
+t.create('missing project')
+  .get('/alerts/g/some-org/this-project-doesnt-exist.json')
+  .expectJSONTypes(Joi.object().keys({
```
Comment (chris48s) [inline services/lgtm/lgtm.tester.js]: A bit of project convention: We use `.expectJSON()` if we're comparing the response to an object literal and `.expectJSONTypes()` if we're checking the type of the response keys or validating against a regex.

File: services/lgtm/lgtm.tester.js (hunk 3 above)
Comment (chris48s) [reply services/lgtm/lgtm.tester.js]: Sorry - ignore this comment. I've just noticed it is already `alerts?$` not `alerts$`.
My bad

File: server.js (hunk 2 above)
Comment (samlanning) [reply server.js]: It indeed is, i'll change it

File: services/lgtm/lgtm.tester.js (hunk 3 above)
Comment (samlanning) [reply services/lgtm/lgtm.tester.js]: :+1: In fairness, i did originally write it with out the `?`, i'll add tests to make sure that it prints it our correctly

**Comment by samlanning** [comment]
Addressed comments :)

File: server.js [hunk 5]
```diff
@@ -1067,6 +1067,38 @@ cache(function(data, match, sendBadge, request) {
...
+  const badgeData = getBadgeData('lgtm', data);
+  request(url, function(err, res, buffer) {
+    if (checkErrorResponse(badgeData, err, res, 'project not found')) {
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      const data = JSON.parse(buffer);
+      if (!('alerts' in data))
+        throw new Error("Invalid data");
```
Comment (RedSparr0w) [inline server.js]: ~~Could you change this to be lower case please `invalid data`~~

Edit: never mind, just realised we aren't using this for output.

**Comment by chris48s** [comment]
nice one - thanks

**Pull request merged**